Post tutorial fixup PR

GiudGiud opened the pull request:
Catches a few things I noticed last week

See https://github.com/idaholab/moose/issues/21745 for the list
I think there's too many dry slides of text. A good next step would be to add some pics to systems

GiudGiud committed:
Fix tutorial makefile to avoid pointing to the wrong moose directory if
your moose directory is not called moose but a moose dir exists near it

GiudGiud committed:
Remove last instances of Modules / FP in doco
refs #22093

GiudGiud committed:
Fix deprecation of picard_custom_pp, closes #18361

GiudGiud committed:
No need for virtual destructors on derived classes. refs #21745
Add more cv qualifiers

GiudGiud committed:
Add mesh diagnostics to troubleshooting
Add a summary slide on classes + templates

GiudGiud committed:
Add examples of PDEs to motivate using FEM
refs #21745

GiudGiud committed:
Add some explanation to AD, refs #21745

GiudGiud committed:
Add Positions system to multiapp tutorial

GiudGiud committed:
Add more explanation about what we re trying to do with multiapps

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1773153/) on f3d4b90 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25484/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1773195/) on f3d4b90 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>49df46</th>
      <th colspan=4>#25484 f3d4b9</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.72%</td>
      <td>85.72%</td>
      <td>-0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>91538</td>
      <td>91535</td>
      <td>-3</td>
      <td>2</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15245</td>
      <td>15249</td>
      <td>+4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25484/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25484/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25484/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud committed:
Allow alerts in darcy tutorial
Add image files needed to config.yml

GiudGiud committed:
Avoid user mistake in restricting sidesets

GiudGiud committed:
Update functor error message for PP being functors

GiudGiud commented on tutorials/darcy_thermo_mech/doc/content/workshop/numerical/fem_overview.md:
```diff
@@ -144,12 +161,16 @@ Generating a weak form generally involves these steps:
     functions and simultaneously generate boundary integrals.
 
+!alert note title=Exercise
+Obtain the weak form for the equations listed before, write the equations in terms of the coefficients
```
```suggestion
Obtain the weak form for the equations listed on the previous slide
```

GiudGiud commented on tutorials/darcy_thermo_mech/doc/content/workshop/numerical/fem_solve.md:
```diff
@@ -246,6 +246,27 @@ often results in decreased solve time overall.
+
+- operator overloading
+
+\equation
```
```suggestion
!equation
```

GiudGiud committed:
Apply suggestions from code review
- spelling & typos
- use reference

Apply suggestions from code review

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

Apply suggestions from code review

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

lindsayad commented on framework/src/executioners/FixedPointSolve.C:
```diff
@@ -492,7 +492,8 @@ FixedPointSolve::computeCustomConvergencePostprocessor()
 
-  auto ppname = getParam<PostprocessorName>("custom_pp");
+  auto ppname = isParamValid("custom_pp") ? getParam<PostprocessorName>("custom_pp")
+                                          : getParam<PostprocessorName>("picard_custom_pp");
```
I can't find this parameter name. Where is it?

GiudGiud replied:
it s in PicardSolve

lindsayad replied:
If you use the relatively new `deprecateParam` API on the `PicardSolve::validParams`, then lookups of "custom_pp" will also consider if the user has set "picard_custom_pp" and then you don't have to check both here (which also has the advantage of not referencing a derived classes parameter name here in the base class)

GiudGiud replied:
this doesnt work as well here because we are deprecating the parameter of the derived class with a parameter of the parent class. 

so if I did that I would allow the deprecated parameter in the other derived classes

lindsayad replied:
right, that's what I was suggesting

GiudGiud replied:
let s see if it works

lindsayad commented on framework/src/meshgenerators/ParsedGenerateSideset.C:
```diff
@@ -103,6 +103,14 @@ ParsedGenerateSideset::ParsedGenerateSideset(const InputParameters & parameters)
   // Handle incompatible parameters
   if (_include_only_external_sides && _check_neighbor_subdomains)
     paramError("include_only_external_sides", "External sides dont have neighbors");
+  const auto included_boundaries = getParam<std::vector<BoundaryName>>("included_boundaries");
```
```suggestion
  const auto & included_boundaries = getParam<std::vector<BoundaryName>>("included_boundaries");
```

lindsayad replied:
this got lost

lindsayad commented on tutorials/darcy_thermo_mech/doc/content/workshop/numerical/fem_overview.md:
```diff
@@ -106,9 +106,27 @@ respective functions.
+FEM is widely applicable for a large range of PDEs and domains.
+
+Example PDEs:
+Have you seen them before? Are they linear/ nonlinear? Coupled?
```
```suggestion
Have you seen them before? Are they linear/nonlinear? Coupled?
```

joshuahansel commented on framework/src/meshgenerators/ParsedGenerateSideset.C:
```diff
@@ -103,6 +103,14 @@ ParsedGenerateSideset::ParsedGenerateSideset(const InputParameters & parameters)
+    paramError("new_boundary",
+               "A boundary cannot be both the new boundary and be included in the list of included "
+               "boundaries. If you are trying to restrict an existing boundary, you must use a "
+               "different name for the new_boundary, then delete the old sideset and rename the "
```
```suggestion
               "different name for 'new_boundary', delete the old boundary, and then rename the "
```

joshuahansel commented on framework/src/meshgenerators/ParsedGenerateSideset.C:
```diff
@@ -103,6 +103,14 @@ ParsedGenerateSideset::ParsedGenerateSideset(const InputParameters & parameters)
+               "A boundary cannot be both the new boundary and be included in the list of included "
+               "boundaries. If you are trying to restrict an existing boundary, you must use a "
+               "different name for the new_boundary, then delete the old sideset and rename the "
+               "new one if you wish to use the old name");
```
```suggestion
               "new boundary to the old boundary.");
```

GiudGiud commented:
@joshuahansel
